pkg/helper/airgap.go

```diff
@@ -0,0 +1,171 @@
+package helper
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+func SetupBastion(cluster *shared.Cluster, flags *customflag.FlagConfig) {
+	shared.LogLevel("info", "Downloading %v artifacts...", cluster.Config.Product)
+	_, err := getArtifacts(cluster, flags)
+	Expect(err).To(BeNil())
+
+	setupPrivateRegistry(cluster, flags)
+	dockerActions(cluster, flags)
+
+	pwd, err := shared.RunCommandOnNode("pwd", cluster.GeneralConfig.BastionIP)
+	Expect(err).To(BeNil())
+
+	regMap := map[string]string{
+		"$PRIVATE_REG": cluster.GeneralConfig.BastionDNS,
+		"$USERNAME":    flags.AirgapFlag.RegistryUsername,
+		"$PASSWORD":    flags.AirgapFlag.RegistryPassword,
+		"$HOMEDIR":     pwd,
+	}
+
+	updateRegistryFile(cluster, regMap)
+}
+
+func setupPrivateRegistry(cluster *shared.Cluster, flags *customflag.FlagConfig) {
+	shared.LogLevel("info", "Setting up private registry...")
+	cmd := fmt.Sprintf(
+		"sudo chmod +x private_registry.sh && "+
+			`sudo ./private_registry.sh "%v" "%v" "%v"`,
+		flags.AirgapFlag.RegistryUsername, flags.AirgapFlag.RegistryPassword,
+		cluster.GeneralConfig.BastionDNS)
+	_, err := shared.RunCommandOnNode(cmd, cluster.GeneralConfig.BastionIP)
+	Expect(err).To(BeNil())
+}
+
+func dockerActions(cluster *shared.Cluster, flags *customflag.FlagConfig) {
+	shared.LogLevel("info", "Running Docker ops script on bastion...")
+	cmd := "sudo chmod +x docker_ops.sh && " +
+		fmt.Sprintf(
+			`sudo ./docker_ops.sh "%v" "%v" "%v" "%v"`,
+			cluster.Config.Product, cluster.GeneralConfig.BastionDNS,
+			flags.AirgapFlag.RegistryUsername, flags.AirgapFlag.RegistryPassword)
+	_, err := shared.RunCommandOnNode(cmd, cluster.GeneralConfig.BastionIP)
+	Expect(err).To(BeNil())
+	shared.LogLevel("info", "Docker pull/tag/push completed!")
+}
+
+func CopyAssetsOnNodes(cluster *shared.Cluster) {
+	shared.LogLevel("info", "Copying assets on all the nodes...")
+	nodeIPs := cluster.ServerIPs
+	nodeIPs = append(nodeIPs, cluster.AgentIPs...)
+
+	var wg sync.WaitGroup
+	for _, nodeIP := range nodeIPs {
+		wg.Add(1)
+		nodeIP := nodeIP
+		go func() {
+			defer wg.Done()
+			copyAssets(cluster, nodeIP)
+			copyRegistry(cluster, nodeIP)
+			makeExecs(cluster, nodeIP)
+		}()
+	}
+	wg.Wait()
+	shared.LogLevel("info", "Copying files complete!")
+}
+
+func copyAssets(cluster *shared.Cluster, ip string) {
+	cmd := fmt.Sprintf(
+		"sudo chmod 400 /tmp/%v.pem && ", cluster.AwsEc2.KeyName)
+	if cluster.Config.Product == "rke2" {
```

Comment:
if you reallyyy want to explicitly validate both products like that, a switch would make it more readable., also adding a default error

Reply:
Updated